datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -2907,6 +3064,46 @@ Projection: a, b
         assert_optimized_plan_eq(plan, expected)
     }
 
+    #[test]
```

Comment:
Assuming other outer join types(Left, Right, Full) covered by existing tests?

Reply:
I think the following existing tests can validate for this change:
- Left: [`filter_using_left_join`](https://github.com/JasonLi-cn/arrow-datafusion/blob/feature/optimize_infer_join_predicate/datafusion/optimizer/src/push_down_filter.rs#L2134)
- Right: [`filter_using_right_join`](https://github.com/JasonLi-cn/arrow-datafusion/blob/feature/optimize_infer_join_predicate/datafusion/optimizer/src/push_down_filter.rs#L2172)
- Full: [`full_join_on_with_filter`](https://github.com/JasonLi-cn/arrow-datafusion/blob/ab2a4f5b701650589dcca052e05aae77b5ccaab6/datafusion/optimizer/src/push_down_filter.rs#L2497)